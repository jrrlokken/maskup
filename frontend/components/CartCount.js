import { TransitionGroup, CSSTransition } from 'react-transition-group';
import styled from 'styled-components';

const Badge = styled.div`
  background-color: var(--black);
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`;

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: all 0.4s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: scale(2) rotateX(0.5turn);
  }
  .count-enter-active {
    transform: rotateX(0);
  }
  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }
  .count-exit {
    transform: scale(2) rotateX(0.5turn);
  }
`;

const CartCount = ({ count }) => (
  <AnimationStyles>
   <TransitionGroup>
     <CSSTransition
      className='count'
      classNames='count'
      key={count}
      timeout={{ enter: 400, exit: 400}}
      unmountOnExit
      >
      <Badge>{count}</Badge>
     </CSSTransition>
   </TransitionGroup>
  </AnimationStyles>
);

export default CartCount;