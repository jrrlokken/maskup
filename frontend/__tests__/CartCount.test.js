import { configure, shallow } from 'enzyme';
import Adapter from '@hteker/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';

import CartCount from '../components/CartCount';

configure({ adapter: new Adapter() });
const count = 10;

describe('<CartCount/>', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(
      <CartCount count={count}/>
    );
    const badge = wrapper.find('CartCount__Badge');
    expect(badge).toMatchSnapshot();
    expect(badge.text()).toBe('10');
  });
});
