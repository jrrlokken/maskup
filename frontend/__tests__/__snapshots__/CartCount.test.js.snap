// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> renders and matches snapshot 1`] = `ShallowWrapper {}`;
